pkg/kots/spec.go

```diff
@@ -78,38 +69,28 @@ func (fs SpecFiles) separate() (SpecFiles, error) {
 	separatedSpecFiles := SpecFiles{}
 
 	for _, file := range fs {
-		if !file.isYAML() || !file.hasContent() {
+		if !file.isYAML() {
 			separatedSpecFiles = append(separatedSpecFiles, file)
 			continue
 		}
 
-		reader := bytes.NewReader([]byte(file.Content))
-		decoder := goyaml.NewDecoder(reader)
-
-		for docIndex := 0; ; docIndex++ {
-			var doc goyaml.MapSlice
-			err := decoder.Decode(&doc)
+		cleanedContent := util.CleanYaml(file.Content)
+		docs := strings.Split(cleanedContent, "\n---\n")
 
-			if err == io.EOF {
-				break
-			} else if err != nil {
-				return nil, errors.Wrap(err, "failed to decode spec file content")
+		for index, doc := range docs {
+			if strings.HasPrefix(doc, "---\n") {
+				doc = doc[4:]
 			}
 
-			var buf bytes.Buffer
-			encoder := goyaml.NewEncoder(&buf)
-			err = encoder.Encode(doc)
-
-			if err != nil {
-				return nil, errors.Wrap(err, "failed to encode separated doc")
+			if len(doc) == 0 {
```

Comment:
should this be `len(strings.TrimSpace(doc))`?

Reply:
`util.CleanYaml` removes all empty lines, so trimming is not needed here

Reply:
doc is not passed through CleanYaml.  it's a result of strings.Split.